elastic/kibana
# [Logs+] Add specialized locators

closes https://github.com/elastic/kibana/issues/158382


## 📝  Summary

This PR adds new customized locators to the log explorer profile. At the moment we only implemented a single dataset selector locator, but we are planning to have more when we start implementing the multiselect in the dataset selector.
The PR replaces the temp navigation to the default discover we implemented for[ 8.10 here](https://github.com/elastic/kibana/pull/163218).

## ✅  Testing

1. Navigate to the onboarding flow `/app/observabilityOnboarding/`
2. Choose either System logs or Stream log files
3. Go through the onboarding wizard
4. Click the Explore logs button at the end
5. You should be redirected to discover in the log explorer profile with the integration and dataset preselected.

## 🎥 Demo


https://github.com/elastic/kibana/assets/11225826/def94fe5-941f-4e88-b871-995442ad9ddc
